BaseTools/Source/Python/Workspace/DscBuildData.py

```diff
@@ -763,6 +763,12 @@ def Modules(self):
                 LibraryPath = PathClass(NormPath(Record[1], Macros), GlobalData.gWorkspace, Arch=self._Arch)
                 LineNo = Record[-1]
 
+                # Validate that the Library instance implements the Library Class
+                if not self._ValidateLibraryClass(LibraryClass, LibraryPath):
+                    EdkLogger.warn("build",

```

Comment:
There was an open earlier about warning visibility from edk2 tools. Are these visible in our current builds?

Reply:
This is correct. RunCmd() does no parsing of the output, only redirecting to the specified log level. The `scan_compiler_output` could be updated to detect Build.py warnings and split them out after the build. It would be the same situation as `path_utilities`. It spews the warning multiple times as end up parsing, and throwing the warning, twice per build. Here is an example of building mu_tiano_platforms:


![image](https://github.com/microsoft/mu_basecore/assets/24388509/f6e52ea2-c88b-441a-9529-1f53908e4e26)

Reply:
I have a small update queued that can fix the spamming, if you want it added.